gossipsub.go

```diff
@@ -206,7 +206,10 @@ type GossipSubParams struct {
 
 // NewGossipSub returns a new PubSub object using the default GossipSubRouter as the router.
 func NewGossipSub(ctx context.Context, h host.Host, opts ...Option) (*PubSub, error) {
-	rt := DefaultGossipSubRouter(h)
+	rt, err := DefaultGossipSubRouter(h)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create default gossipsub router: %w", err)
+	}
 	opts = append(opts, WithRawTracer(rt.tagTracer))
```

Comment:
We need to find a way to absorb this (the tag tracer) into the defaults, it is very error prone to require it to be explicitly instantaited.